src/utilities/permissions.ts

```diff
@@ -625,6 +625,9 @@ const queryPermissions = {
       (getPermission(queries, user) && getRolePlanBranchPermission(queries, user, sourcePlan, targetPlan, model))
     );
   },
+  RESTORE_ACTIVITY_FROM_CHANGELOG: (user: User | null): boolean => {
+    return isUserAdmin(user) || getPermission(['restoreActivityFromChangelog'], user);
```

Comment:
I think this might need to duplicate the permission logic of `UPDATE_ACTIVITY_DIRECTIVE`.